api/STYLE.md

```diff
@@ -141,12 +141,10 @@ To add an extension config to the API, the steps below should be followed:
    (and to not break the docs build).
    See the [key-value-store PR](https://github.com/envoyproxy/envoy/pull/17745/files) for an example of adding a new extension point to common.
 1. Make sure your proto imports the v3 extension config proto (`import "udpa/annotations/status.proto";`)
-1. Make sure your proto is either tracked as a work in progress
-   (`option (udpa.annotations.file_status).work_in_progress = true;`)
-   or ready to be used
+1. Make sure your proto is tracked as ready to be used
    (`option (udpa.annotations.file_status).package_version_status = ACTIVE;`).
-   This is required to automatically include the config proto in [api/versioning/BUILD](versioning/BUILD).
-1. Add a reference to the v3 extension config in (1) in [api/versioning/BUILD](versioning/BUILD) under `active_protos`.
+   This is required to automatically include the config proto in [api/versioning/BUILD](versioning/BUILD) under `active_protos`, and without it `proto_format.sh` will delete the proto file.
+1. Add a reference to the v3 extension config in (2) in [api/BUILD](BUILD) under `v3_protos`. Without this, `proto_format.sh` will erase any imports of your proto file from other proto files.
```

Comment:
I agree with what you are writing, just trying to get some clarity on why the line was erased.
IIRC this is what happened, but feel free to correct me.

You introduced 2 new protos, lets call them A and B, where B imported A.
When you ran `proto_format` it removed the import line from B.
Was the line deleted because the A wasn't in the BUILD file or was it because B wasn't there (or both).
I'm assuming that it was because A wasn't declared in the `api/BUILD`.

If one would first add extension A, run the script, and only then add B and run the script, would it still erase the import line in B?

Reply:
Actual sequence:
I added proto A, and ran `proto_format`. `api/BUILD` remains untouched.
I added proto B, importing A, and ran `proto_format` again. The import of A was removed.
I undid that change.
I added proto A to api/BUILD, and ran `proto_format` again. The import of A is not removed.

Nothing actually goes wrong (I believe CI even passes) if you don't add proto A to `api/BUILD`, until another proto tries to depend on it (that was even in a separate PR in this case). `api/BUILD` doesn't get automatically populated at all, because the thing that *would* automatically populate it doesn't see the thing to add unless you already added it. So I think it's essentially just acting as an auto-formatter of that build file, not an auto-populater. (Similar to how you have to add the proto's build dependencies to the "do not edit" BUILD file before the script will run and "auto-generate" that BUILD file - but worse because for *that* case you get an error message, vs. for `api/BUILD` there's not even a warning that stuff wasn't generated).

Reply:
Thanks.
For some reason I was under the impression the `proto_format` will add the proto to the `api/BUILD` file, but if it doesn't, then the comment in `api/BUILD` is wrong (not only needs to be clarified).

Reply:
The comment in every api/**/v3/BUILD file is wrong in the same way. They all say `# DO NOT EDIT. This file is generated by tools/proto_format/proto_sync.py.`, but you have to edit them manually if you want to add an import or proto_format.sh won't run.

I definitely would rather not include that change into this PR, since changing the generator output will also require updating every generated-ish file it touches.

Reply:
I think it's not really wrong for `api/**v3/BUILD`. You have to populate this file once, initially, when you create a new package. After that point it's automatically maintained and not changed. I think we should be discouraging editing after the initial checkin in most of these files (that's the point of the warning).